### PR TITLE
refactor: button component

### PR DESCRIPTION
**Pull request by PwFaze**
## Change made

-   [ ] Bug fixes
-   [ ] New features
-   [x] Breaking changes

## Describe what you have done

-

### New Features

-

### Fix

-

### Others

-

### CODE REVIEW

File: .env.example
Comment (shalluv) [inline .env.example:1]: Please do not delete this file.

Comment (shalluv) [inline .env.example:1]: Please do not delete this file.

File: src/components/DefaultButton.tsx [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import React from 'react';
+
+const DefaultButton = ({ content, additionalStyle, onClick, disabled }) => {
+    const buttonClassName =
+        `max-w-full w-40 mx-auto items-center py-2 text-white transition-all duration-500 hover:ring-8 disabled:cursor-not-allowed disabled:ring-0 ` +
```
Comment (shalluv) [inline src/components/DefaultButton.tsx]: There is no need to explicitly declare a variable, just write to the `className` attribute of the `button`.

File: src/components/DefaultButton.tsx
Comment (shalluv) [inline src/components/DefaultButton.tsx:1]: Simply call it "Button". The term "Default button" can be misleading.

**Review by shalluv — CHANGES_REQUESTED** [review]
Please take a look at `clsx`. I think it will make it easier to construct `className`.

File: src/components/DefaultButton.tsx (hunk 1 above)
Comment (leomotors) [reply src/components/DefaultButton.tsx]: You can also use `import clsx from 'clsx'` to concatenate class.

File: src/components/Profile/BaanSelection/components/GroupPanel.tsx [hunk 2]
```diff
@@ -139,13 +140,21 @@ export default function GroupPanel() {
...
+                />
+                {/* <button
                     className="mx-auto rounded-lg bg-pink-400 px-3 py-2 text-xl text-white ring-4 ring-pink-400/30 transition-all duration-500 enabled:hover:ring-8 disabled:cursor-not-allowed disabled:bg-pink-300"
                     onClick={() => openModal('modal-leave-group')}
                     disabled={!isAuthenticated || group?.leaderID === user?.id}
                 >
                     ออกจากกลุ่ม
-                </button>
+                </button> */}
             </div>
```
Comment (leomotors) [inline src/components/Profile/BaanSelection/components/GroupPanel.tsx]: Uncomment and delete these after finished

File: src/components/Button.tsx [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import React from 'react';
+
+const Button = ({ content, additionalStyle, onClick, disabled }) => {
```
Comment (boomchanotai) [inline src/components/Button.tsx]: Please add typescript for this component's props

File: src/context/AuthContext.tsx [hunk 4]
```diff
@@ -46,6 +46,7 @@ const AuthProvider = ({ children }: { children: ReactNode }) => {
         const token = localStorage.getItem('token');
         if (token) {
             const userProfile = await getUserProfile();
+            console.log(userProfile);
```
Comment (boomchanotai) [inline src/context/AuthContext.tsx]: Please delete console.log if it not necessary

**Review by boomchanotai — CHANGES_REQUESTED** [review]
Cool! Please add typescript for button for Button Component

**Review by boomchanotai — APPROVED** [review]
Great Job!

**Review by shalluv — APPROVED** [review]

**Pull request merged**